### PR TITLE
Adding managed preferences class

### PR DESCRIPTION
**Pull request by khawkins**
Starting with the wrapper class for managed preferences.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ The associated labels for the provided login hosts.
+ */
+@property (nonatomic, readonly) NSArray *loginHostLabels;
+
+/**
+ The raw NSDictionary of managed preferences.
+ */
+@property (nonatomic, strong, readonly) NSDictionary *rawPreferences;
+
+@end
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.h]: On Android, we also have remote access consumer key and oauth redirect url (to allow runtime replacement for the values in bootconfig.json). See https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/unstable/libs/SalesforceSDK/src/com/salesforce/androidsdk/config/RuntimeConfig.java

**Comment by wmathurin** [comment]
Looks fine otherwise.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.h (hunk 1 above)
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.h]: @akhileshgupta had told me to leave those settings out.  Akhilesh, should we not have parity across platforms for these?

Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.h]: @wmathurin I went ahead and added those in.  I also added a notification listener for changes to the settings that can be pushed in at runtime.

Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFManagedPreferences.h]: (https://github.com/khawkins/SalesforceMobileSDK-iOS/commit/5a9ff6f1f80a2437074f1ae0cc38c61be1dd67f9)

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**